service/controller/resource/app/desired.go

```diff
@@ -215,7 +209,7 @@ func (r *Resource) newApp(appOperatorVersion string, cr apiv1beta1.Cluster, appS
 	if key.IsBundle(appName) {
 		appName = fmt.Sprintf("%s-%s", key.ClusterID(&cr), appName)
 		desiredAppOperatorVersion = key.UniqueOperatorVersion
-		appNamespace = key.ClusterID(&cr)
+		appNamespace = "giantswarm"
```

Comment:
The `appNamespace` seems now to be used in both fields, `.metadata.namespace` and `.spec.namespace` which is wrong. We cannot set the `.spec.namespace` field to the `giantswarm` value for bundle apps, for then the nested App CRs will get deployed to the `giantswarm` namespace, while we still want them in the cluster namespace. Hence we want to have something like this:

```yaml
apiVersion: application.giantswarm.io/v1alpha1
kind: App
metadata:
  labels:
    app-operator.giantswarm.io/version: 0.0.0
  name: <cluster_id>-observability-bundle
  namespace: giantswarm
spec:
  catalog: default
  config:
    configMap:
      name: <cluster_id>-cluster-values
      namespace: <cluster_id>
  kubeConfig:
    ...
    inCluster: true
    ...
  name: observability-bundle
  namespace: <cluster_id>
```